Source/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSMachHeaders.h

```diff
@@ -39,9 +39,16 @@ typedef struct {
 static BSG_Mach_Binary_Images bsg_mach_binary_images;
 
 /**
- * A lock, used to synchronise access to the array of binary image info.
+ * An OS-version-specific lock, used to synchronise access to the array of binary image info.
  */
-static os_unfair_lock bsg_mach_binary_images_access_lock = OS_UNFAIR_LOCK_INIT;
+#if defined(__IPHONE_10_0) || defined(__MAC_10_12) || defined(__TVOS_10_0) || defined(__WATCHOS_3_0)
+    #import <os/lock.h>
+    static os_unfair_lock bsg_mach_binary_images_access_lock = OS_UNFAIR_LOCK_INIT;
+#else
+    #import <libkern/OSAtomic.h>
+    #define OS_SPINLOCK_INIT 0
```

Comment:
This seems to be defined in the header already, does this statement need to be included for functional reasons or can it be removed?

Reply:
Hah.   Missed due to the `#if` blocking symbol resolution.